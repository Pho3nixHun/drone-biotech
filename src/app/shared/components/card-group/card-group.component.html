<div
    class="grid xl:grid-cols-[1fr_1fr_1.5fr_1fr_1fr_1fr_1fr_1fr_1fr] gap-4 xl:gap-y-1 md:grid-cols-2 text-left duration-500 transition-all rounded-md py-3"
>
    <div
        class="xl:display hidden xl:grid grid-cols-subgrid col-span-full bg-gray-300 border border-gray-400 p-3 tracking-wider"
    >
        <ng-content select="div" />
    </div>
    <ng-content select="app-card-item" />
</div>
