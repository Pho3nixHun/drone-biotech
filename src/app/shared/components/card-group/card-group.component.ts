import { Component } from '@angular/core';

@Component({
    selector: 'app-card-group',
    imports: [],
    templateUrl: './card-group.component.html'
})
export class CardGroupComponent {}
