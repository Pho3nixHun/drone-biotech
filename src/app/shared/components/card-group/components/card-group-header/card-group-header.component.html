<header
    class="xl:display hidden xl:grid grid-cols-subgrid col-span-full bg-gray-300 border border-gray-400 tracking-wider *:flex *:items-center p-3"
>
    <ng-content select="div" />
</header>
