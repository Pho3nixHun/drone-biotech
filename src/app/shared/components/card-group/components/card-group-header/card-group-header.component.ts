import { Component } from '@angular/core';

@Component({
    selector: 'app-card-group-header',
    imports: [],
    templateUrl: './card-group-header.component.html',
})
export class CardGroupHeaderComponent {}
