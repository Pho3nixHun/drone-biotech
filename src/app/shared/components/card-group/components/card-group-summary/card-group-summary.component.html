<div class="flex flex-col space-y-3">
    <ng-content
        select="[app-card-group-summary-header, app-card-group-summary-content]"
    />
</div>
