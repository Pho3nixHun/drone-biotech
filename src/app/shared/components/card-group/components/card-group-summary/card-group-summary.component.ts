import { Component } from '@angular/core';

@Component({
    selector: 'app-card-group-summary',
    imports: [],
    templateUrl: './card-group-summary.component.html',
})
export class CardGroupSummaryComponent {}
