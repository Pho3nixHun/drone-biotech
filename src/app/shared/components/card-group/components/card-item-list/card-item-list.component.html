<div class="grid grid-cols-1 lg:grid-cols-2 xl:grid-cols-subgrid col-span-full">
    <ng-content select="app-card-item" />
</div>
