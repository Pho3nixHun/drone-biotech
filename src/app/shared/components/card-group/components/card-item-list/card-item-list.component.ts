import { Component } from '@angular/core';

@Component({
    selector: 'app-card-item-list',
    imports: [],
    templateUrl: './card-item-list.component.html',
})
export class CardItemListComponent {}
