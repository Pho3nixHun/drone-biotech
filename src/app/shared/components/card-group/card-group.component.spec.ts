import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CardGroupComponent } from './card-group.component';
import { Component } from '@angular/core';
import { CardItemListComponent } from './components/card-item-list/card-item-list.component';
import { CardGroupSummaryComponent } from './components/card-group-summary/card-group-summary.component';
import { CardGroupHeaderComponent } from './components/card-group-header/card-group-header.component';

@Component({
    imports: [
        CardGroupHeaderComponent,
        CardGroupComponent,
        CardItemListComponent,
        CardGroupSummaryComponent,
    ],
    template: `
        <app-card-group>
            <app-card-group-header></app-card-group-header>
            <app-card-item-list></app-card-item-list>
            <app-card-group-summary></app-card-group-summary>
            <div>Should not be projected</div>
        </app-card-group>
    `,
})
class TestHostComponent {}

describe('CardGroupComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
