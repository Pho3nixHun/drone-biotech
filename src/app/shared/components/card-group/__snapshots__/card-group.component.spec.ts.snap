// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGroupComponent should project the content and render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-card-group>
    <div
      class="grid grid-cols-subgrid col-span-full gap-3"
    >
      <app-card-group-header>
        <header
          class="xl:display hidden xl:grid grid-cols-subgrid col-span-full bg-gray-300 border border-gray-400 tracking-wider *:flex *:items-center p-3"
        />
      </app-card-group-header>
      <app-card-item-list>
        <div
          class="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-subgrid col-span-full gap-3"
        />
      </app-card-item-list>
      <app-card-group-summary>
        <div
          class="flex flex-col space-y-3"
        />
      </app-card-group-summary>
    </div>
  </app-card-group>
</div>
`;
